modules/ROOT/pages/policy-mule4-detokenization.adoc

```diff
@@ -73,51 +85,55 @@ If the detokenized value for the Social Security Number in the example were _365
 }
 --
 
-You can configure the detokenization policy to detokenize values in your payload's request or response.
+You can configure the detokenization policy to detokenize values in the request or response of your payload.
 
-To learn how to tokenize a value refer to the xref:policy-mule4-tokenization.adoc[Tokenization Policy].
+=== Detokenize Requests
 
-=== Detokenizing Requests
+The following events occur when detokenizing requests:
 
 image::policy-mule4-tokenization-67b25.png[]
 
 . A user performs a request to the application protected by the policy.
-. The request is first caught by Anypoint Security, and redirected to the application, reaching first the detokenization policy.
-. The detokenization policy is configured to detokenize the request, so it extracts all the tokens from the payload using the selector expression.
-. The detokenization service receives all the tokens, applies the transformation back to sensitive data, and then returns the result to the policy.
-. The policy replaces the payload with the detokenized values, and the request reaches the Mule application.
+. The request goes first to Anypoint Security and then to the application Detokenization policy.
+. The Detokenization policy extracts all tokens from the payload that match the selector expression.
+. The detokenization service receives the tokens, transforms them back to sensitive data, and returns the result to the policy.
+. The policy replaces the payload with the detokenized values and forwards it to the Mule application.
 . The Mule application processes the request and returns the result.
 
 === Detokenizing Responses
 
+The following events occur when detokenizing responses:
+
 image::policy-mule4-tokenization-9347d.png[]
 
 . A user performs a request to the application protected by the policy.
-. The request is first caught by Anypoint Security, and redirected to the application, reaching first the detokenization policy.
-. The detokenization policy is configured to detokenize the Mule application's response, so it redirects the request without extracting anything from the payload.
+. The request goes first to Anypoint Security and then to the application Detokenization policy.
+. The Detokenization policy tranforms the response of the Mule application by redirecting the request without extracting anything from the payload.
 . The Mule application processes the request and returns the result.
-. The detokenization policy extracts all the tokens from the payload using the selector expression.
-. The tokenization service receives all the tokens, applies the transformation back to sensitive data, and then returns the result to the policy.
-. The policy sends the response back to the user, with the detokenized values in the payload.
+. The detokenization policy extracts the tokens from the payload using the selector expression.
+. The tokenization service receives the tokens, transforms them back to sensitive data, and  returns the result to the policy.
```

Comment:
The tokenization service or the detokenization service?
```suggestion
. The tokenization service receives the tokens, returns them to their original sensitive values, and sends the result to the policy.
```

Reply:
Done! Thanks!